components/file_limit_sticky_banner/index.tsx

```diff
@@ -0,0 +1,185 @@
+
+import OverlayTrigger from 'components/overlay_trigger';
+import Tooltip from 'components/tooltip';
+
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {get as selectPreference} from 'mattermost-redux/selectors/entities/preferences';
+import {isCurrentLicenseCloud, getSubscriptionProduct as selectSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import Constants, {CloudProducts, Preferences} from 'utils/constants';
+import {GlobalState} from 'types/store';
+import useGetUsage from 'components/common/hooks/useGetUsage';
+import {fallbackStarterLimits} from 'utils/limits';
+import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
+import NotifyAdminCTA from 'components/notify_admin_cta/notify_admin_cta';
+import {FileSizes} from 'utils/file_utils';
+
+interface FileLimitSnoozePreference {
+    lastSnoozeTimestamp: number;
+}
+
+const snoozeCoolOffDays = 10;
+const snoozeCoolOffDaysMillis = snoozeCoolOffDays * 24 * 60 * 60 * 1000;
+
+const StyledDiv = styled.div`
+width: 100%;
+padding: 0 24px;
+margin: 12px auto;
+`;
+
+const InnerDiv = styled.div`
+position: relative;
+border: 1px solid #FFBC1F;
+border-radius: 4px;
+background: linear-gradient(0deg, rgba(255, 212, 112, 0.16), rgba(255, 212, 112, 0.16)), #FFFFFF;
+padding: 16px 17px;
+color: var(--center-channel-color);
+`;
+
+const CloseIcon = styled.button`
+border: none;
+outline: none;
+background: none;
+position: absolute;
+top: 16px;
+right: 0;
+`;
+
+const StyledI = styled.i`
+color: #FFBC1F;
+`;
+
+function FileLimitStickyBanner() {
+    const [show, setShow] = useState(true);
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+
+    const usage = useGetUsage();
+    const openPricingModal = useOpenPricingModal();
+
+    const user = useSelector(getCurrentUser);
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const product = useSelector(selectSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+
+    const snoozePreferenceVal = useSelector((state: GlobalState) => selectPreference(state, Preferences.CLOUD_USER_EPHEMERAL_INFO, 'file_limit_banner_snooze'));
+
+    let shouldShowAgain = true;
+    if (snoozePreferenceVal !== '') {
+        const snoozeInfo = JSON.parse(snoozePreferenceVal) as FileLimitSnoozePreference;
+        const timeDiff = Date.now() - snoozeInfo.lastSnoozeTimestamp;
+        shouldShowAgain = timeDiff >= snoozeCoolOffDaysMillis;
+    }
+
+    if (!show) {
+        return null;
+    }
+
+    if (!shouldShowAgain) {
+        return null;
+    }
+
+    if (!isCloud || !isStarter) {
+        return null;
+    }
+
+    let isAbovePlanFileStorageLimit = false;
+
+    const currentFileStorageUsage = usage.files.totalStorage;
```

Comment:
Any chance `usage` can be null that may require a null check here?

Reply:
Changed to `const currentFileStorageUsage = usage?.files?.totalStorage || 0;`